### DISCUSSION TITLE
How can I release an approved iOS submission

### DISCUSSION THREAD

**Discussion by kyurkchyan** [fastlane/fastlane#21867] [Q&A]
We have a 3-step release process
1. Submit bundle of the app to TestFlight
2. Submit the bundle to app store for review (without automatic release)
3. Once the app is approved, we manually release the app when we feel it's the right time. 

Our submission fastlane script looks like this

```
lane :submit_for_app_store_review do |options|
  deliver(
    api_key_path: options[:api_key_path],
    app_identifier: options[:bundle_id],
    build_number: options[:version_code],
    app_version: options[:version_name],
    metadata_path: options[:metadata_path],
    precheck_include_in_app_purchases: false,
    skip_screenshots: true,
    skip_binary_upload: true,
    skip_metadata: false,
    automatic_release: false,
    force: true,
    submission_information: {
      add_id_info_uses_idfa: false
    },
    submit_for_review: true,
    reject_if_possible: true
  )
end
```

As you can see `automatic_release` is set to false.

Once the app is approved, we go to the app store connect page and manually release the app from there. We want to automate this step. I need help finding documentation about how manual release can be achieved for already approved submissions. It's not a big deal, but it'd be cool to have the full automation done from CI/CD instead of having to visit app store connect for this last step.

**Comment by robinmacharg** [comment]
I added this to our Fastlane setup recently. Our situation is slightly more complicated in that I manage 12 React Native apps across both Android and iOS and an app release involves a lot more repetitive clicking. For releasing the app I dig into the `Spaceship::ConnectAPI` package directly and make use of helper functionality provided there. The iOS portion of the `release_apps` lane looks something like (the meat of the answer is towards the bottom):

```ruby
app = Spaceship::ConnectAPI::App.find(bundle_id)

# Get the status from the App Store
live = app.get_live_app_store_version
edit = app.get_edit_app_store_version
latest = app.get_latest_app_store_version
pending_release = app.get_pending_release_app_store_version # You're interested in this one
in_review = app.get_in_review_app_store_version

# Logging and release
if !latest.nil?
  log "Latest version: v#{latest.version_string}"
end

if !live.nil?
  log "Current live version: v#{live.version_string}"
else
  log "#{name} is not currently live"
end

if !edit.nil? && edit.app_store_state != Spaceship::ConnectAPI::AppStoreVersion::AppStoreState::ACCEPTED
  log "#{name} (v#{edit.version_string}) has not been accepted (currrent state: #{edit.app_store_state})"
end

if !in_review.nil?
  log "#{name} (v#{in_review.version_string}) is currently in review"
else
  log "#{name} is not currently in review"
end

if !pending_release.nil?
  log "#{name} is pending release."

  # This next line does the release
  Spaceship::ConnectAPI.post_app_store_version_release_request(app_store_version_id: pending_release.id)

else
  log "#{name} does not have a version pending release.  Skipping release."
end
```

There's a custom `log` function that I'll not patronise the answer with (`puts` will do fine). HTH.
